src/librustc/ty/layout.rs

```diff
@@ -1935,7 +1935,7 @@ impl<'tcx> LayoutOf for LayoutCx<'tcx, TyCtxt<'tcx>> {
     /// Computes the layout of a type. Note that this implicitly
     /// executes in "reveal all" mode.
     fn layout_of(&self, ty: Ty<'tcx>) -> Self::TyLayout {
-        let param_env = self.param_env.with_reveal_all();
+        let param_env = self.param_env.with_reveal_all_normalized(self.tcx);
```

Comment:
This can be done lazily, only if `ty.has_projections()`.

Reply:
I'm not 100% sure that this is correct - is there any `Reveal:All` end up influencing things when the type has no projections?